crowbar_framework/app/models/service_object.rb

```diff
@@ -1576,4 +1616,28 @@ def profile(name, &block)
       block.call
     end
   end
+
+  def equal_or_match(value, maybe_regexp)
+    operator = lambda do
+      op, check = maybe_regexp.split(" ", 2)
+      value.to_f.send(op.to_sym, check.to_f)
+    end
+    
+    case
+    when maybe_regexp.start_with?('/') && maybe_regexp.end_with?('/')
+      Regexp.new(maybe_regexp[1..-2]).match(value)
+    when maybe_regexp.start_with?('==')
+      operator.call
+    when maybe_regexp.start_with?('>=')
+      operator.call
+    when maybe_regexp.start_with?('>')
+      operator.call
+    when maybe_regexp.start_with?('<=')
+      operator.call
+    when maybe_regexp.start_with?('<')
+      operator.call
+    else
+      value == maybe_regexp
```

Comment:
I do not see the problem with '=='. Also many version like '10.10' do not behave correctly for the to_f.to_s implicit conversion.

What is the main point of removing the explicit '==' for numeric versions?  I would prefer not to remove this and avoid implicit conversions hidden from the developer.